state/src/font_style.rs

```diff
@@ -162,12 +191,63 @@ impl State<CustomAttributeValues> for FontStyle {
     }
 }
 
-fn parse_font_style(style: &str) -> skia_safe::FontStyle {
+fn parse_font_style(style: &str) -> Slant {
     match style {
-        "italic" => skia_safe::FontStyle::italic(),
-        "bold" => skia_safe::FontStyle::bold(),
-        "bold-italic" => skia_safe::FontStyle::bold_italic(),
-        _ => skia_safe::FontStyle::default(),
+        "upright" => Slant::Upright,
+        "italic" => Slant::Italic,
+        "oblique" => Slant::Oblique,
+        _ => Slant::Upright,
+    }
+}
+
+fn parse_font_weight(weight: &str) -> Weight {
+    // NOTES:
+    // This is mostly taken from the OpenType specification (https://learn.microsoft.com/en-us/typography/opentype/spec/os2#usweightclass)
+    // CSS has one deviation from this spec, which uses the value "950" for extra_black.
+    // skia_safe also has an "invisible" weight smaller than the thin weight, which could fall under CSS's interpretation of OpenType's
+    // version. In this case it would be font_weight: "50".
+    match weight {
+        "invisible" => Weight::INVISIBLE,
+        "thin" => Weight::THIN,
+        "extra-light" => Weight::EXTRA_LIGHT,
+        "light" => Weight::LIGHT,
+        "normal" => Weight::NORMAL,
+        "medium" => Weight::MEDIUM,
+        "semi-bold" => Weight::SEMI_BOLD,
+        "bold" => Weight::BOLD,
+        "extra-bold" => Weight::EXTRA_BOLD,
+        "black" => Weight::BLACK,
+        "extra-black" => Weight::EXTRA_BLACK,
+        // "50" => skia_safe::font_style::Weight::INVISIBLE,
+        "100" => Weight::THIN,
+        "200" => Weight::EXTRA_LIGHT,
+        "300" => Weight::LIGHT,
+        "400" => Weight::NORMAL,
+        "500" => Weight::MEDIUM,
+        "600" => Weight::SEMI_BOLD,
+        "700" => Weight::BOLD,
+        "800" => Weight::EXTRA_BOLD,
+        "900" => Weight::BLACK,
+        // "950" => Weight::EXTRA_BLACK,
```

Comment:
Shouldn't we enable both this one and the `// "50" => skia_safe::font_style::Weight::INVISIBLE,` ?

Reply:
Probably. I left it commented because its not really something in the OpenType standard, but CSS does use weights of 950 (not 50 though skia does let you). I'll uncomment though since it makes sense and I wanted some outside feedback.